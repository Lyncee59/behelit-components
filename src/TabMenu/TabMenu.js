import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { prop } from '../Tools/interpolation'

const Wrapper = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: ${prop('width')};
  margin-bottom: 0.5rem;
`

const TabMenu = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

TabMenu.propTypes = {
  children: PropTypes.node.isRequired,
  width: PropTypes.string,
}

TabMenu.defaultProps = {
  width: '100%'
}

export default TabMenu
