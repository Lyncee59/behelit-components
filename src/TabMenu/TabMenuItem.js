import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Text } from '../Typography'
import { palette, theme } from '../Tools/interpolation' 

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  font-family: ${theme('fontPrimary')};
  font-size: 1rem;
  font-weight: 400;
  color: ${(props) => props.selected ? props.theme['tabMenuItemSelectedColor'] : props.theme['tabMenuItemColor']};
  background: ${(props) => props.selected ? props.theme['tabMenuItemSelectedBackgroundColor'] : props.theme['tabMenuItemBackgroundColor']};
  border: 1px solid ${(props) => props.selected ? props.theme['tabMenuItemSelectedBorderColor'] : props.theme['tabMenuItemBorderColor']};
  cursor: pointer;
`

const TabMenuItem= ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

TabMenuItem.propTypes = {
  children: PropTypes.node.isRequired,
  selected: PropTypes.bool,
}

TabMenuItem.defaultProps = {
  selected: false
}

export default TabMenuItem
