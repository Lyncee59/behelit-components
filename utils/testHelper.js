import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)