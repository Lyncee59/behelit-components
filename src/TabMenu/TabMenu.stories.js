import React from 'react'
import { storiesOf } from '@storybook/react'

import TabMenu from './TabMenu'
import TabMenuItem from './TabMenuItem'

storiesOf('TabMenu', module)
  .add('default', () => (
    <TabMenu>
      <TabMenuItem>First item</TabMenuItem>
      <TabMenuItem>Second item</TabMenuItem>
      <TabMenuItem>Third item</TabMenuItem>
    </TabMenu>
  ))
  .add('with selected', () => (
    <TabMenu>
      <TabMenuItem>First item</TabMenuItem>
      <TabMenuItem selected>Second item</TabMenuItem>
      <TabMenuItem>Third item</TabMenuItem>
    </TabMenu>
  ))