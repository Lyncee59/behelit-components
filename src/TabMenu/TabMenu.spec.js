import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import TabMenu from './TabMenu'

describe('TabMenu', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TabMenu>TabMenu</TabMenu>)

    expect(tree.find(TabMenu)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<TabMenu width="300px">TabMenu</TabMenu>)

    expect(tree.find(TabMenu)).toMatchSnapshot()
  })
})
