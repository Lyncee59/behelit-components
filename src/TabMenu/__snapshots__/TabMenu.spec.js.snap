// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-bottom: 0.5rem;
}

<TabMenu
  width="100%"
>
  <styled.div
    width="100%"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: ",
              [Function],
              ";
  margin-bottom: 0.5rem;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      width="100%"
    >
      <div
        className="c0"
        width="100%"
      >
        TabMenu
      </div>
    </StyledComponent>
  </styled.div>
</TabMenu>
`;

exports[`TabMenu should render correctly with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
  margin-bottom: 0.5rem;
}

<TabMenu
  width="300px"
>
  <styled.div
    width="300px"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: ",
              [Function],
              ";
  margin-bottom: 0.5rem;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      width="300px"
    >
      <div
        className="c0"
        width="300px"
      >
        TabMenu
      </div>
    </StyledComponent>
  </styled.div>
</TabMenu>
`;
